lib/index.js

```diff
@@ -1,24 +1,33 @@
 'use strict'
 
 const { join } = require('path')
-const { readJson } = require('../util/read')
+const { read } = require('../util/read')
 const { convertYarnToNpmV1, convertNpmV1ToYarn } = require('./lockfileV1')
 const { convertYarnToNpmV2, convertNpmV2ToYarn } = require('./lockfileV2')
 
+const checkWorkspace = (requiresWorkspace, withWorkspace) => {
+  if (requiresWorkspace && !withWorkspace) {
+    throw new Error('Workspace (npm lockfile v2) support is experimental. Pass `--with-workspaces` flag to enable and cross your fingers. Good luck!')
+  }
+
+  return requiresWorkspace
+}
+
 module.exports = {
-  convertYarnToNpm (yarnLock, name, version, packageDir) {
-    const packageJson = readJson(join(packageDir, 'package.json'))
+  convertYarnToNpm (yarnLock, name, version, packageDir, withWorkspace) {
+    const packageJson = read(join(packageDir, 'package.json'))
+    const requiresWorkspace = !!packageJson.workspaces
```

Comment:
Any chance we could do without this? If the user uses the flag, let's deal with workspaces, otherwise, let's not.